### ISSUE TITLE
  Feature Request: Forwarding remaining balance to next month

### ISSUE THREAD

**Issue by mjfelecio** [emavgl/oinkoin#167]
Hi, I have been an avid user of your app for 3 months now. Your app has been the simplest and best expense tracker ever. 

However, I really think it could benefit from having an option in the setting to forward the balance to next month. 

Having to transfer the balance to the next month by making it an expense and putting it to the next month through income is really ugly. Currently I use the yearly view instead but that is too broad to properly see my expenses and income. I hope you consider this

**Comment by emavgl** [comment]
Hi! :) I am happy you like the app.

In the settings there is the option to choose what to visualise by default in the homepage. For example, if you want to see your all balance all the time, you can set it to load all records.

Does it work for you?

What do you mean by "too broad". Maybe you want to keep in the homepage records of the months but visualise in the overview box the overall balance?

**Comment by mjfelecio** [comment]
Thanks for taking the time to understand my feature request. Let me try to explain the problem I'm facing in more detail and why an automatic balance forwarding feature would be so helpful.

The main issue is that I need a reliable way to carry over my monthly balances without having to manually create clunky "expense" and "income" transactions. Currently, if I have $1,400 remaining at the end of October, the only way for me to transfer that to November is to create an "expense" transaction to zero out the October balance, and then an "income" transaction to add $1,400 to November. This creates a few problems:

1. It clutters up my transaction history, making it harder to get a clear picture of my actual monthly expenses and income.
2. It introduces the potential for human error, as I could accidentally forget to make the balancing transactions or make mistakes in the amounts.
3. It makes it difficult to get an accurate, at-a-glance understanding of my overall financial situation. When viewing the yearly summary, the artificially inflated income and expense numbers make it hard to quickly assess my true financial health.

What I'm hoping for is an option in the settings to automatically "forward" my ending balance from one month to the beginning balance of the next month. This would solve all of these issues by:

1. Keeping my monthly transaction history clean and accurate.
2. Eliminating the risk of manual data entry errors.
3. Providing a truer representation of my finances in the yearly summary and other high-level views, especially in monthly views.

Does this help explain my feature request a bit better? I'm new to programming so I am not sure about the complexity of this task so I'm happy to clarify or provide any additional details. Please let me know if you have any other questions!

**Comment by emavgl** [comment]
Thanks for expanding. While I understand the issue and what you want to solve, it is not clear to me why the actual solutions: "Setting > Homepage time interval" does not solve the issue. Is it because you want to keep in the homepage just the records of the months, but you want to see the overall balance instead?

**Comment by mjfelecio** [comment]
Thank you for continuing the discussion. Let me walk through what I'm seeing to make sure we're on the same page:

When I use "All Records" view:
- I can see my total income (30,466), total expenses (22,226), and accurate overall balance (8,240)
- This gives me the most accurate financial picture because it includes money carried over from previous months
- However, it shows ALL transactions, making it difficult to focus on my current month's spending patterns

When I use "Records of the current month":
- I only see this month's transactions: income (8,000), expenses (1,010), balance (6,990)
- This helps me focus on current month's activities
- But it doesn't account for the money I had from previous months, so the balance isn't reflecting my true available funds

What I think would be ideal is:
1. Being able to view ONLY the current month's transactions (for clarity and focus)
2. While still seeing my true total balance that includes carried-over funds

I noticed you mentioned something about "seeing the overall balance instead". It sounds like it might be exactly what I'm looking for if it means I could see my current month's transactions while still having visibility of my total available balance.

**Comment by emavgl** [comment]
Thanks for taking the time to explain in detail. Yes, 

-    Being able to view ONLY the current month's transactions (for clarity and focus)
-    While still seeing my true total balance that includes carried-over funds

This is what I meant by `seeing the overall balance instead`. It is definitely possible to do that, I'll update you when I find the time to do that :)

**Comment by Rooyca** [comment]
This feature would be awesome! :smile:

**Comment by emavgl** [comment]
I just implemented it, if you want, you can give it a try with https://github.com/emavgl/oinkoin/actions/runs/13092806210#artifacts
I'll appreciate your feedback.

Please, backup your data before trying it.

**Comment by Rooyca** [comment]
First of all, thank you very much for taking the time to design this app, excellent work. The new version seems to be working correctly; the only thing is that to apply the changes, it is necessary to close and reopen the app. Other than that, everything is fine.

**Comment by emavgl** [comment]
Yeah, unfortunately restart is needed at the moment. Avoiding the restart is not at the moment an easy change and it requires architectural changes. Anyway, I think it is not a big issue, since it is a change you do once.

The release is https://github.com/emavgl/oinkoin/releases/tag/1.0.79

I am closing this issue. Thanks for providing the feedback and trying this out.
